src/gateway/bridge/mod.rs

```diff
@@ -97,9 +98,16 @@ pub struct ShardRunnerInfo {
 /// Newtype around a callback that will be called on every incoming request. As long as this
 /// collector should still receive events, it should return `true`. Once it returns `false`, it is
 /// removed.
-pub struct CollectorCallback(pub Box<dyn Fn(&Event) -> bool + Send + Sync>);
+#[derive(Clone)]
+pub struct CollectorCallback(pub Arc<Box<dyn Fn(&Event) -> bool + Send + Sync>>);
```

Comment:
Arc does not require T to be sized, so you can store `Arc<dyn Fn(&Event) + ...>`

Reply:
This should incidentally solve your type complexity lint issue.

Reply:
Will update that.